From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 24 Nov 2022 01:19:12 +0100
Subject: [PATCH] SIMD support

License: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following patch:
"Add SIMD utilities"
By: Kevin Raneri <kevin.raneri@gmail.com>
As part of: Pufferfish (https://github.com/pufferfish-gg/Pufferfish)
Licensed under: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 5720bf7133f7a7fe81993e70f178fc0675bcde40..3d64e5564e6317a12a5bfd8ba2861da6d7cc6958 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -267,6 +267,13 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         org.dreeam.leaf.command.LeafCommands.registerCommands(this); // Leaf - Leaf commands
         com.destroystokyo.paper.VersionHistoryManager.INSTANCE.getClass(); // Paper - load version history now
 
+        // Gale start - Pufferfish - SIMD support
+        // Initialize vectorization
+        try {
+            gg.pufferfish.pufferfish.simd.SIMDDetection.initialize();
+        } catch (Throwable ignored) {}
+        // Gale end - Pufferfish - SIMD support
+
         // this.worldData.setGameType(properties.gameMode.get()); // CraftBukkit - moved to world loading
         LOGGER.info("Default game type: {}", properties.gameMode.get());
         // Paper start - Unix domain socket support
